cloud/vmware/vsphere_guest.py

```diff
@@ -689,6 +692,10 @@ def deploy_template(vsphere_client, guest, resource_pool, template_src, esxi, mo
                 cloneArgs["linked"] = True
                 cloneArgs["snapshot"] = snapshot_to_clone
 
+            if vm_extra_config.get("folder") is not None:
```

Comment:
@marvinpinto Good catch!  I tested and an empty string results in the same behavior as None, which is that the VM is cloned into the same folder as the template.  So I would say that yes the feature does degrade gracefully.